### ISSUE TITLE
[Bug]: Qwen3-VL-2B-Instruct produces significantly different (degraded) outputs on v0.12.0 compared to v0.11.0

### ISSUE THREAD

**Issue by AGENDD** [vllm-project/vllm#36117]
### Your current environment

```
- vLLM v0.11.0 (baseline) and v0.12.0 (regression)
- Model: Qwen/Qwen3-VL-2B-Instruct
- GPU: 4x A100, tensor-parallel-size=4
- Evaluation: PointBench (982 samples, point-in-mask coordinate localization)
- Parameters: temperature=0.0, top_p=0.9, max_tokens=1024
```

### 🐛 Describe the bug

When serving **Qwen3-VL-2B-Instruct** with vLLM, upgrading from **v0.11.0 to v0.12.0** causes a severe degradation in coordinate localization outputs. The model produces systematically biased coordinates, resulting in accuracy dropping from **~60% to ~13%** on PointBench.

This is **not** a scoring/evaluation issue — the raw model outputs are fundamentally different between versions for identical inputs and parameters.

### Observed behavior

The model's predicted coordinates shift dramatically between versions:

| Metric | v0.11.0 | v0.12.0 |
|--------|---------|---------|
| **PointBench accuracy** | **57.6%** | **~13%** |
| X coordinate mean | 509 | 285 |
| Y coordinate mean | 573 | 1074 |
| Coordinates exceeding 0-1000 range | 0% | 10% |

On v0.12.0, coordinates are **systematically biased** — X values skew low (toward left edge) and Y values skew high (often exceeding the valid 0-1000 range).

### Visual evidence

Side-by-side comparison on identical inputs (left: v0.11.0, right: v0.12.0). Blue dots = correct (hit mask), Red dots = incorrect (miss mask), Green overlay = ground truth mask area.

**Example 1: "Point to the handle to pick up the mug"**
- v0.11.0: `[520, 250]` → correctly on the handle ✓
- v0.12.0: `[85, 174]` → top-left corner, completely wrong ✗

![compare_5](https://raw.githubusercontent.com/AGENDD/vllm-issue-assets/main/images/compare_5.png)

**Example 2: "Point to the front wheels of the car"**
- v0.11.0: `[354, 718], [544, 718]` → both wheels correctly identified ✓
- v0.12.0: `[100, 700]` → single point, mislocated ✗

![compare_10](https://raw.githubusercontent.com/AGENDD/vllm-issue-assets/main/images/compare_10.png)

**Example 3: "Point to 'coins'"**
- v0.11.0: `[500, 838]` → correct ✓
- v0.12.0: `[171, 825]` → offset to the left ✗

![compare_4](https://raw.githubusercontent.com/AGENDD/vllm-issue-assets/main/images/compare_4.png)

**More comparisons:**

![compare_7](https://raw.githubusercontent.com/AGENDD/vllm-issue-assets/main/images/compare_7.png)

![compare_9](https://raw.githubusercontent.com/AGENDD/vllm-issue-assets/main/images/compare_9.png)

### Root cause analysis

We compared the source code of `vllm/model_executor/models/qwen3_vl.py` between v0.11.0 and v0.12.0 and identified the following changes that likely contribute to this regression:

#### 1. `nn.Conv3d` → `Conv3dLayer` (most likely cause)

The `PatchEmbed` layer's projection changed from `nn.Conv3d` to `Conv3dLayer` (CustomOp from PR #28455). On PyTorch 2.9, `Conv3dLayer.forward_cuda` uses `F.linear` (unfold + matmul) instead of `F.conv3d`:

```python
# v0.11.0
self.proj = nn.Conv3d(in_channels, hidden_size, kernel_size=kernel_size, stride=kernel_size, bias=True)

# v0.12.0
self.proj = Conv3dLayer(in_channels, hidden_size, kernel_size=kernel_size, stride=kernel_size, bias=True)
```

While mathematically equivalent when `kernel_size == stride`, the different computation paths produce numerical differences in bf16/fp16 that accumulate through 28 ViT layers.

#### 2. `apply_rotary_pos_emb_vision` precision path change

```python
# v0.11.0: converts to float32 internally, computes cos/sin from raw freqs
def apply_rotary_pos_emb_vision(t, freqs):
    t_ = t.float()
    cos = freqs.cos()
    sin = freqs.sin()
    ...

# v0.12.0: receives pre-computed cos/sin
def apply_rotary_pos_emb_vision(t, cos, sin):
    rotary_emb_function = dispatch_rotary_emb_function(
        default=partial(apply_rotary_emb_torch, is_neox_style=True))
    ...
```

#### 3. Attention backend changes

v0.12.0 removed xFormers backend support and refactored attention wrappers for `torch.compile` compatibility.

#### 4. Position embedding interpolation

`fast_pos_embed_interpolate` bilinear weight calculation changed from `w00 = 1 - dh - dw + w11` to `w00 = 1 - dh - w01` — mathematically equivalent but different FP computation order.

### Reproduction

```python
import requests, base64

def test(url, image_b64, prompt):
    resp = requests.post(f"{url}/v1/chat/completions", json={
        "model": "inference",
        "messages": [
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": [
                {"type": "image_url", "image_url": {"url": f"data:image/jpeg;base64,{image_b64}"}},
                {"type": "text", "text": prompt}
            ]}
        ],
        "temperature": 0.0, "max_tokens": 1024, "top_p": 0.9
    })
    return resp.json()["choices"][0]["message"]["content"]

# Deploy same model with vllm==0.11.0 and vllm==0.12.0
# vllm serve "Qwen/Qwen3-VL-2B-Instruct" --tensor-parallel-size 4 --port 3204

# Use any image and coordinate-based prompt
# v0.11.0 produces coordinates centered around the target
# v0.12.0 produces coordinates biased toward top-left with Y values often > 1000
```

### Expected behavior

Both vLLM versions should produce equivalent outputs for the same model, input, and parameters.

### Related issues

- #27406 — MM performance regression from PyTorch 2.9 (Conv3d)
- #27418 — Replace PatchEmbed's conv3d to linear layer (workaround)
- #28455 — Extract conv layer as CustomOp (introduced Conv3dLayer)
- #33204 — Qwen3-VL-Embedding produces different embeddings than official implementation

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
